### PR TITLE
Enable Travis testing

### PR DESCRIPTION
**Pull request by alan-knight**

### CODE REVIEW

File: test/notification_test.dart [hunk 1]
```diff
@@ -33,15 +33,17 @@ main() {
...
-      var expectation = Notification.supported ? returnsNormally : throws;
-      expect(() {
-        allDefaults = new Notification("Hello world");
-        allSpecified = new Notification("Deluxe notification",
-            dir: "rtl",
-            body: 'All parameters set',
-            icon: 'icon.png',
-            tag: 'tag',
-            lang: 'en-US');
+        var expectation = Notification.supported ? returnsNormally : throws;
```
Comment (a14n) [inline test/notification_test.dart]: Shouldn't it be indented by +2 instead of +4 ?

File: test/notification_test.dart [hunk 2]
```diff
@@ -55,14 +57,13 @@ main() {
...
     });
   });
 }
 
 /// Simulate clicking on the notification.
-clickOn(thing) {
+clickOn(Notification thing) {
   var event =
-      new MouseEvent('click', view: thing, canBubble: true, cancelable: true);
+      new MouseEvent('click', canBubble: true, cancelable: true);
```
Comment (a14n) [inline test/notification_test.dart]: It can fit in one line.

File: test/notification_test.html [hunk 3]
```diff
@@ -3,10 +3,13 @@
...
+   <link rel="x-dart-test" href="notification_test.dart">
+<!--   <script src="notification_test.dart" type="application/dart"></script>
+   <script src="packages/browser/dart.js"></script> -->
+
 </head>
 <body>
-<button id="b">Click me</button>
-<script src="notification_test.dart" type="application/dart"></script>
-<script src="packages/browser/dart.js"></script>
+    <button id="b">Click me</button>
```
Comment (a14n) [inline test/notification_test.html]: This button is not used in your test. So I would say `test/notification_test.html` can be removed.

File: tool/travis.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+# Fast fail the script on failures.   
+set -e
+
+pub run test -p content-shell
```
Comment (a14n) [inline tool/travis.sh]: You can also test on firefox on travis. See https://gist.github.com/a14n/441101c997ee8f935d9c#file-travis-yml-L10

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+language: dart
+with_content_shell: true
+script: ./tool/travis.sh
```
Comment (a14n) [inline .travis.yml]: You can directly put `pub run test ....` here and skip `tools/travis.sh`. See https://gist.github.com/a14n/441101c997ee8f935d9c

File: test/notification_test.dart (hunk 1 above)
Comment (alan-knight) [inline test/notification_test.dart]: Ran formatter

File: test/notification_test.dart (hunk 2 above)
Comment (alan-knight) [inline test/notification_test.dart]: Same

File: .travis.yml (hunk 5 above)
Comment (alan-knight) [inline .travis.yml]: Cool. Done.

File: tool/travis.sh (hunk 4 above)
Comment (alan-knight) [inline tool/travis.sh]: Even more cool. Done.

File: test/notification_test.html (hunk 3 above)
Comment (alan-knight) [inline test/notification_test.html]: Yeah, ok. Makes it a bit more work to debug standalone.

**Pull request merged**